### PR TITLE
Remove master branch merges from LTS template

### PR DESCRIPTION
**Pull request by NotMyFault**
Content on the master branch should go through the same backporting process, like any other changes that land on stable branches.
If there's no change, that's eligible for a backport, the branches should not be merged.

### CODE REVIEW

**Review by timja — CHANGES_REQUESTED** [review]
To remove this is needs to be clarified to check the repositories for backporting needs the steps currently only mention the Jenkins repo from what I can see

**Comment by NotMyFault** [comment]
That's part of this step: https://github.com/jenkins-infra/release/blob/master/.github/ISSUE_TEMPLATE/1-lts-release-checklist.md?plain=1#L45

packaging uses Jira and therefore has its LTS candidates there, for this repository I added an entry.

File: .github/ISSUE_TEMPLATE/1-lts-release-checklist.md [hunk 1]
```diff
@@ -42,7 +42,8 @@ This role should rotate between LTS releases
 
 - [ ] Backport changes, create a local branch in jenkinsci/jenkins, run the [list-issue-commits script](https://github.com/jenkins-infra/backend-commit-history-parser/blob/master/bin/list-issue-commits) to locate commits via jira ID, some manual work is required to locate them if the issue ID wasn't present at merge time, backport with `git cherry-pick -x $commit`
 
-- [ ] Open backporting PR with into-lts label and summary of changes in description from [lts-candidate-stats script](https://github.com/jenkins-infra/backend-commit-history-parser/blob/master/bin/lts-candidate-stats) and the selected [Jira lts-candidates](https://issues.jenkins-ci.org/issues/?filter=12146)
+- [ ] Open backporting PR with into-lts label and summary of changes in description from [lts-candidate-stats script](https://github.com/jenkins-infra/backend-commit-history-parser/blob/master/bin/lts-candidate-stats) and the selected [Jira lts-candidates](https://issues.jenkins-ci.org/issues/?filter=12146)  
+Also check the [release repository](https://github.com/jenkins-infra/release/issues?q=is%3Aclosed+label%3Alts-candidate+) for additional candidates.
```
Comment (timja) [inline .github/ISSUE_TEMPLATE/1-lts-release-checklist.md:46]: And packaging

Comment (NotMyFault) [reply .github/ISSUE_TEMPLATE/1-lts-release-checklist.md]: Oic, if packaging uses GH issues, it'd make sense to archive the Jira, no?

I've added the packaging repo, just need to create the label, I have no permission there.

**Review by timja — APPROVED** [review]

**Pull request merged**